src/main/java/core/basesyntax/fileservice/FileParseImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.fileservice;
+
+import core.basesyntax.model.FruitBatch;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileParseImpl implements FileParse {
+    @Override
+    public List<FruitBatch> parseList(List<List<String>> listOfLists) {
+        if (listOfLists == null) {
+            throw new IllegalArgumentException("List is null");
```

Comment:
Do we have to throw NullPointer here?

Reply:
I am not sure myself but from the description of IllegalArgumentException it seems to fit here better:
```
Thrown to indicate that a method has been passed an illegal or inappropriate argument.
```
While description of NullPointerException says:
```
Thrown when an application attempts to use null in a case where an object is required.
```